src/domain/noteSettings.service.ts

```diff
@@ -99,4 +107,36 @@ export default class NoteService {
   public async patchMemberRoleByUserId(id: NoteId, userId: UserId, newRole: MemberRole): Promise<MemberRole> {
     return await this.noteSettingsRepository.patchMemberRoleByUserId(id, userId, newRole);
   }
+
+  /**
+   * Set new parent for the note
+   *
+   * @param id - Note id
+   * @param parentURL - link to the new parent note
+   */
+  public async updateParent(id: NoteId, parentURL: string): Promise<void> {
+    // Regex matches any characters between '/note/' and the next slash or end of string
+    const regex = /\/note\/([^/]+)/;
+
+    const matches = parentURL.match(regex);
+
+    if (matches === null) {
+      throw new Error('Invalid parent URL');
+    }
+
+    // Extracts the ID from the URL. The ID is matches[1] as matches[0] is the full match
+    const parentId = matches[1];
+
+    const noteIdPattern = /^[a-zA-Z0-9-_]{10}$/;
```

Comment:
why don't to put this regex to the first regex?

Reply:
Because in the case where there is no slash at the end, we will take the first 10 characters of what comes after “/note/”. This may cause unexpected behavior
For example, when ```parentURL = '/note/1234567890abc'``` match will be like this ``` match = '1234567890' ```. This could be the id of a real note that we don't want to be the parent of